components/api-resource-mgt/org.wso2.carbon.identity.api.resource.collection.mgt/src/main/java/org/wso2/carbon/identity/api/resource/collection/mgt/APIResourceCollectionManager.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.resource.collection.mgt;
+
+import org.wso2.carbon.identity.api.resource.collection.mgt.exception.APIResourceCollectionMgtException;
+import org.wso2.carbon.identity.api.resource.collection.mgt.model.APIResourceCollection;
+import org.wso2.carbon.identity.api.resource.collection.mgt.model.APIResourceCollectionSearchResult;
+
+import java.util.List;
+
+/**
+ * API Resource Collection Manager Interface.
+ */
+public interface APIResourceCollectionManager {
+
+    /**
+     * Get API Resource Collections.
+     *
+     * @param filter Filter expression.
+     * @return API resource collection search result.
+     * @throws APIResourceCollectionMgtException If an error occurred while retrieving API Resource Collections.
+     */
+    APIResourceCollectionSearchResult getAPIResourceCollections(String filter) throws APIResourceCollectionMgtException;
+
+    /**
+     * Get API Resource Collections with required attributes.
+     *
+     * @param filter             Filter expression.
+     * @param requiredAttributes Required attributes.
+     * @return API resource collection search result with required attributes.
+     * @throws APIResourceCollectionMgtException If an error occurred while retrieving API Resource Collections.
+     */
+    APIResourceCollectionSearchResult getAPIResourceCollectionsWithRequiredAttributes(String filter,
```

Comment:
Why do we need two methods to fetch API resource collections? Can't we use a single method to fetch api resource collections in both cases

Reply:
```suggestion
    APIResourceCollectionSearchResult getAPIResourceCollections(String filter,
```

Reply:
Removed one.